crates/core/src/idx/planner/iterators.rs

```diff
@@ -556,8 +557,59 @@ impl IndexRangeThingIterator {
 		db: DatabaseId,
 		ix: &DefineIndexStatement,
 	) -> Result<Self> {
-		let range = full_iterator_range();
-		Self::new(irf, ns, db, ix, &range)
+		Self::new(irf, ns, db, ix, &full_iterator_range())
+	}
+
+	pub(super) fn compound_range(
+		irf: IteratorRef,
+		ns: NamespaceId,
+		db: DatabaseId,
+		ix: &IndexReference,
+		prefix: &[Value],
+		ranges: &[(BinaryOperator, Arc<Value>)],
+	) -> Result<Self> {
+		let (from, to) = Self::reduce_range(ranges)?;
+		Ok(Self {
+			irf,
+			r: Self::range_scan_prefix(ns, db, ix, prefix, from, to)?,
+		})
+	}
+
+	/// Determines the lowest and highest values in the range
+	fn reduce_range(ranges: &[(BinaryOperator, Arc<Value>)]) -> Result<(RangeValue, RangeValue)> {
+		let mut from = vec![];
+		let mut to = vec![];
+		for (op, v) in ranges {
+			let key = storekey::serialize(v.as_ref())?;
+			match op {
+				BinaryOperator::LessThan => to.push((key, 0, v.clone())),
+				BinaryOperator::LessThanEqual => to.push((key, 1, v.clone())),
+				BinaryOperator::MoreThan => from.push((key, 1, v.clone())),
+				BinaryOperator::MoreThanEqual => from.push((key, 0, v.clone())),
```

Comment:
I think using booleans here instead of the integers would make the code a bit easier to understand. The way the integers are being used confused me until I read the code below to find out that they are essentially booleans for whether or not it was inclusive.